api/v1alpha1/mustgather_types.go

```diff
@@ -56,6 +56,25 @@ type MustGatherSpec struct {
 	// If set to true, resources will be retained. If false or not set, resources will be deleted (default behavior).
 	// +kubebuilder:default:=false
 	RetainResourcesOnCompletion bool `json:"retainResourcesOnCompletion,omitempty"`
+
+	// The image to use for the must-gather pods.
+	// This can currently only be one of two values:
+	// 'default' for most cases including OSD/ROSA, which will use quay.io/openshift/must-gather
+	// 'acm_hcp' for HCP must gathers, which will use registry.redhat.io/multicluster-engine/must-gather-rhel9:v2.8
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum={"default","acm_hcp"}
+	// +kubebuilder:default:=default
+	MustGatherImage string `json:"mustGatherImage,omitempty"`
+
+	// The namespace of the hosted cluster to collect the must-gather for.
+	// This field is required if mustGatherImage='acm_hcp'.
+	// +kubebuilder:validation:Optional
+	HostedClusterNamespace string `json:"hostedClusterNamespace,omitempty"`
+
+	// The name of the hosted cluster to collect the must-gather for.
+	// This field is required if mustGatherImage='acm_hcp'.
+	// +kubebuilder:validation:Optional
+	HostedClusterName string `json:"hostedClusterName,omitempty"`
```

Comment:
Do we think that we'll ever need to add additional options specific to HCP must gathers? I'm not close-enough to the functionality to know if we will or not. 

If so - should we put it into its own subfield to keep the api cleaner?

So it would be something like `HostedClusterOptions` that's a `HostedClusterOptions` type, which would then contain `Namespace` and `Name` string fields.

It would end up looking like this:

```
spec:
  mustGatherImage: acm_hcp
  hostedClusterOptions:
    Namespace: my_hcp_namespace
    Name: my_cluster_name
```